encoder-otel-brave/src/test/java/zipkin2/reporter/otel/brave/ITZipkinSenderTests.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2024 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.reporter.otel.brave;
+
+import static org.assertj.core.api.BDDAssertions.then;
+
+import brave.Span;
+import brave.Span.Kind;
+import brave.Tags;
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import brave.handler.MutableSpan;
+import brave.handler.SpanHandler;
+import brave.propagation.ThreadLocalCurrentTraceContext;
+import brave.propagation.TraceContext;
+import brave.sampler.Sampler;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import okhttp3.OkHttpClient;
+import okhttp3.OkHttpClient.Builder;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import zipkin2.reporter.AsyncReporter;
+import zipkin2.reporter.Encoding;
+import zipkin2.reporter.okhttp3.OkHttpSender;
+
+@Testcontainers(disabledWithoutDocker = true)
+class ITZipkinSenderTests {
+
+  private static final Logger log = LoggerFactory.getLogger(ITZipkinSenderTests.class);
+
+  private static final int EXPECTED_TRACE_SIZE = 5;
+
+  @Container
+  JaegerAllInOne jaegerAllInOne = new JaegerAllInOne();
+
+  JaegerTestingScenario testingScenario;
+
+  @Test
+  void shouldSendBraveSpansToJaegerOtlpEndpoint() throws Exception {
+    testingScenario = new JaegerTestingScenario(
+        jaegerAllInOne.getHttpOtlpPort());
+
+    List<String> traceIds = testingScenario.exportedTraceIds();
+
+    Awaitility.await().untilAsserted(() -> {
+      then(traceIds).hasSize(EXPECTED_TRACE_SIZE);
+      thenAllTraceIdsPresentInBackend(testingScenario.queryUrl(), traceIds);
```

Comment:
one note I didn't mention was that this test isn't sufficient as an integration test as it doesn't verify the data was readable except its trace ID. I think I earlier said you could use an opentelemetry utility to deserialize a trace read back to verify mappings worked as we expected. So, in the test read-back the json such that the reader of the test will see if there is a miss. This is also why I suggest using a constant trace as it makes the verification idempotent, just subject to readback delay.

Meanwhile, I'll poke around to see if otel has a first party utility in the collector to read-back, or if everyone are using jaeger for this (which is possible)

Reply:
This class is replaced by [ITOtelEncoderTest](https://github.com/openzipkin-contrib/zipkin-otel/pull/7/files#diff-f1614b313ae6e7ebe173556465333def6f893b421571522e9bb496a42ede83c3)